ledger-tool/src/main.rs

```diff
@@ -1653,6 +1758,21 @@ fn main() {
                             })
                             .ok();
                     }
+
+                    if let Some(recorded_slots_file) = record_slots_file {
+                        if let Ok(recorded_slots) = recorded_slots.clone().unwrap().lock() {
+                            let bank_hashes =
+                                bank_hash_details::BankHashDetails::new(recorded_slots.to_vec());
+
+                            // writing the json file ends up with a syscall for each number, comma, indentation etc.
+                            // use BufWriter to speed things up
+
+                            let writer = std::io::BufWriter::new(recorded_slots_file);
+
+                            serde_json::to_writer_pretty(writer, &bank_hashes).unwrap();
```

Comment:
This is duplicate code with stuff we have in `bank_hash_details.rs`, would be nice to single source that

Reply:
`fn write_bank_hash_details_file()` write a single bank, and here we are writing a `Vec<BankHashDetails>`. Also `fn write_bank_hash_details_file()` is used from `fn dump_then_repair_correct_slots()` as well. 

I am not sure how I can consolidate the two.

Reply:
Hmm yeah, maybe not as easy as I thought. I'd say we can leave it for now

The long term vision is implementing something like `BankHashDetailsStreamer`. This object could hold the serializer, and then dump out details bank-by-bank. Will definitely help out if someone runs the command with "accounts" mode on a large number of slots.

You could specify the config at the start, the object could retain the config. Then, when you call `stream_bank()` or whatever we decide to name it, the streamer extracts appropriate details (just the bank hash OR all the hashes + accounts).

To the end user, this is mostly invisible tho. So, like I said, I'm happy to push the PR as-is and then potentially go refactor after the fact

Reply:
Sure, that makes sense. For virtual machine debugging it would be super-useful to have the transactions in the file too. Let start with this and go from there